src/core/context.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Charm Context definition and parsing logic."""
+
+from enum import Enum
+
+from charms.data_platform_libs.v0.data_interfaces import RequirerData
+from ops import ActiveStatus, BlockedStatus, CharmBase, MaintenanceStatus, Relation
+
+from common.utils import WithLogging
+from core.domain import S3ConnectionInfo
+
+S3 = "s3-credentials"
```

Comment:
nit: could this live probably in `constants.py` as well?

Reply:
Done